{"failure_commit": "07748a9d4115e208b19470462e76c1ab38b7693c", "repair_commit": "a7b9fa866579a080bc52c6d091f755345907a4d4", "error_lines": ["crypto/dh/dh_backend.c:85:38: error: too few arguments to function call, expected 2, have 1"], "workflow_id": 16615429536, "job_id": [47006937763, 47006937812, 47006938017, 47006938028], "workflow_name": "GitHub CI", "job_name": ["memory_sanitizer", "freebsd-x86_64", "out-of-readonly-source-and-install-macos (macos-13)", "out-of-readonly-source-and-install-macos (macos-14)"], "diffs": [{"error_line": "crypto/dh/dh_backend.c:85:38: error: too few arguments to function call, expected 2, have 1", "relevant_repairs": []}], "repair_source": "main_branch", "error_details": ["crypto/dh/dh_backend.c:85:38: error: too few arguments to function call, expected 2, have 1\n   85 |         if (ossl_dh_check_pairwise(dh) == 0)\n      |             ~~~~~~~~~~~~~~~~~~~~~~   ^\ninclude/crypto/dh.h:45:5: note: 'ossl_dh_check_pairwise' declared here\n   45 | int ossl_dh_check_pairwise(const DH *dh, int return_on_null_numbers);\n      |     ^                      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~"], "compilation_related_paths": {"direct_includes": ["include/crypto/dh.h"], "indirect_includes": ["include/crypto/dh.h"]}, "compilation_related_paths_details": [{"direct_includes": ["include/crypto/dh.h"], "indirect_includes": ["include/crypto/dh.h"]}]}
{"failure_commit": "90d5d27fcfbce4bd1253d3220c6a5232d234b40a", "repair_commit": "e7d5398aa1349cc575a5b80e0d6eb28e61cb4bfa", "error_lines": ["crypto/cms/cms_asn1.c:240:73: error: expected ';' after top level declarator"], "workflow_id": 16472317719, "job_id": [46564483430, 46564483505, 46564483513, 46564483515, 46564483548, 46564483646, 46564483805, 46564483828], "workflow_name": "GitHub CI", "job_name": ["basic_clang", "no-shared-macos (macos-13)", "freebsd-x86_64", "threads_sanitizer", "no-shared-macos (macos-14)", "memory_sanitizer", "out-of-readonly-source-and-install-macos (macos-13)", "out-of-readonly-source-and-install-macos (macos-14)"], "diffs": [{"error_line": "crypto/cms/cms_asn1.c:240:73: error: expected ';' after top level declarator", "relevant_repairs": []}], "repair_source": "main_branch", "error_details": ["crypto/cms/cms_asn1.c:240:73: error: expected ';' after top level declarator\n  240 | } ASN1_ADB_END(CMS_OtherRecipientInfo, 0, oriType, 0, &ori_def_tt, NULL)\n      |                                                                         ^\n      |                                                                         ;"], "compilation_related_paths": {"error_file": ["crypto/cms/cms_asn1.c"], "direct_includes": ["crypto/cms/cms_local.h"]}, "compilation_related_paths_details": [{"error_file": ["crypto/cms/cms_asn1.c"], "direct_includes": ["crypto/cms/cms_local.h"]}]}
{"failure_commit": "4bdc1981f96a70c8f97f735e953c075d17840ca1", "repair_commit": "61e1905a7a22f690446df63007e6dccdb53af146", "error_lines": ["ssl/quic/quic_ackm.c:1037:12: error: conflicting types for 'ossl_ackm_new'; have 'OSSL_ACKM *(OSSL_TIME (*)(void *), void *, OSSL_STATM *, const OSSL_CC_METHOD *, OSSL_CC_DATA *, char)' {aka 'struct ossl_ackm_st *(OSSL_TIME (*)(void *), void *, struct ossl_statm_st *, const struct ossl_cc_method_st *, struct ossl_cc_data_st *, char)'}"], "workflow_id": 16455404553, "job_id": [46511078053, 46511078080, 46511078083, 46511078086, 46511078129, 46511078142, 46511078146, 46511078152, 46511078153, 46511078197, 46511078199, 46511078210, 46511078254, 46511078264, 46511078269, 46511078273, 46511078276], "workflow_name": "GitHub CI", "job_name": ["check-c99", "basic_gcc", "linux-arm64", "no-shared-ubuntu", "fuzz_tests", "no-legacy", "address_ub_sanitizer", "legacy", "no-deprecated", "enable_non-default_options", "non-caching", "full_featured", "out-of-readonly-source-and-install-ubuntu", "external-tests-misc", "external-tests-pkcs11-provider", "external-tests-oqs-provider", "external-tests-pyca (3.9)"], "diffs": [{"error_line": "ssl/quic/quic_ackm.c:1037:12: error: conflicting types for 'ossl_ackm_new'; have 'OSSL_ACKM *(OSSL_TIME (*)(void *), void *, OSSL_STATM *, const OSSL_CC_METHOD *, OSSL_CC_DATA *, char)' {aka 'struct ossl_ackm_st *(OSSL_TIME (*)(void *), void *, struct ossl_statm_st *, const struct ossl_cc_method_st *, struct ossl_cc_data_st *, char)'}", "relevant_repairs": []}], "repair_source": "dependency", "error_details": [" 1037 | OSSL_ACKM *ossl_ackm_new(OSSL_TIME (*now)(void *arg),\n      |            ^~~~~~~~~~~~~"], "compilation_related_paths": {"error_file": ["ssl/quic/quic_ackm.c"]}, "compilation_related_paths_details": [{"error_file": ["ssl/quic/quic_ackm.c"]}]}
{"failure_commit": "4bdc1981f96a70c8f97f735e953c075d17840ca1", "repair_commit": "61e1905a7a22f690446df63007e6dccdb53af146", "error_lines": ["ssl/quic/quic_ackm.c:1037:12: error: conflicting types for 'ossl_ackm_new'"], "workflow_id": 16455404553, "job_id": [46511078091, 46511078134, 46511078155, 46511078159, 46511078164, 46511078201, 46511078255, 46511078261], "workflow_name": "GitHub CI", "job_name": ["no-shared-macos (macos-13)", "no-shared-macos (macos-14)", "memory_sanitizer", "basic_clang", "threads_sanitizer", "freebsd-x86_64", "out-of-readonly-source-and-install-macos (macos-14)", "out-of-readonly-source-and-install-macos (macos-13)"], "diffs": [{"error_line": "ssl/quic/quic_ackm.c:1037:12: error: conflicting types for 'ossl_ackm_new'", "relevant_repairs": []}], "repair_source": "dependency", "error_details": ["ssl/quic/quic_ackm.c:1037:12: error: conflicting types for 'ossl_ackm_new'\nOSSL_ACKM *ossl_ackm_new(OSSL_TIME (*now)(void *arg),\n           ^\ninclude/internal/quic_ackm.h:22:12: note: previous declaration is here\nOSSL_ACKM *ossl_ackm_new(OSSL_TIME (*now)(void *arg),\n           ^"], "compilation_related_paths": {"error_file": ["ssl/quic/quic_ackm.c"]}, "compilation_related_paths_details": [{"error_file": ["ssl/quic/quic_ackm.c"]}]}
{"failure_commit": "e9e56e513ff54e9a3da168845ea94b9e30fcb3e2", "repair_commit": "491fe4c6d06eadb288da5ee0ee75bfb311aade36", "error_lines": ["ssl/s3_lib.c:4490:5: error: 'alg_k' undeclared (first use in this function); did you mean 'alg_a'?"], "workflow_id": 16373729801, "job_id": [46268326721, 46268326769], "workflow_name": "GitHub CI", "job_name": ["enable_non-default_options", "minimal"], "diffs": [{"error_line": "ssl/s3_lib.c:4490:5: error: 'alg_k' undeclared (first use in this function); did you mean 'alg_a'?", "relevant_repairs": []}], "repair_source": "dependency", "error_details": ["ssl/s3_lib.c:4490:5: error: 'alg_k' undeclared (first use in this function); did you mean 'alg_a'?\n 4490 |     alg_k = s->s3.tmp.new_cipher->algorithm_mkey;\n      |     ^~~~~\n      |     alg_a\nssl/s3_lib.c:4490:5: note: each undeclared identifier is reported only once for each function it appears in\nAt top level:\ncc1: note: unrecognized command-line option '-Wno-unterminated-string-initialization' may have been intended to silence earlier diagnostics"], "compilation_related_paths": {"error_file": ["ssl/s3_lib.c"]}, "compilation_related_paths_details": [{"error_file": ["ssl/s3_lib.c"]}]}
{"failure_commit": "d98c44f116490189453ebd91b1a0d0a3f1f21041", "repair_commit": "6c108b9e4590903a94347cb572d24bc8ebe2c753", "error_lines": ["crypto/threads_pthread.c:747:60: error: 'thread_contention_data' undeclared (first use in this function)", "crypto/threads_pthread.c:749:40: error: invalid application of 'sizeof' to incomplete type 'struct stack_traces'", "crypto/threads_pthread.c:752:11: error: invalid use of undefined type 'struct stack_traces'", "crypto/threads_pthread.c:792:60: error: 'thread_contention_data' undeclared (first use in this function)", "crypto/threads_pthread.c:794:40: error: invalid application of 'sizeof' to incomplete type 'struct stack_traces'", "crypto/threads_pthread.c:797:11: error: invalid use of undefined type 'struct stack_traces'"], "workflow_id": 16050554668, "job_id": [45291945129, 45291945131, 45291945156, 45291945169, 45291945172, 45291945174, 45291945195, 45291945196, 45291945201, 45291945249, 45291945273, 45291945278, 45291945290, 45291945314, 45291945325, 45291945326, 45291945335], "workflow_name": "GitHub CI", "job_name": ["non-caching", "no-deprecated", "check-c99", "linux-arm64", "address_ub_sanitizer", "minimal", "basic_gcc", "fuzz_tests", "no-shared-ubuntu", "no-legacy", "legacy", "out-of-readonly-source-and-install-ubuntu", "external-tests-oqs-provider", "external-tests-pyca (3.9)", "full_featured", "external-tests-misc", "external-tests-pkcs11-provider"], "diffs": [{"error_line": "crypto/threads_pthread.c:747:60: error: 'thread_contention_data' undeclared (first use in this function)", "relevant_repairs": []}, {"error_line": "crypto/threads_pthread.c:749:40: error: invalid application of 'sizeof' to incomplete type 'struct stack_traces'", "relevant_repairs": []}, {"error_line": "crypto/threads_pthread.c:752:11: error: invalid use of undefined type 'struct stack_traces'", "relevant_repairs": []}, {"error_line": "crypto/threads_pthread.c:792:60: error: 'thread_contention_data' undeclared (first use in this function)", "relevant_repairs": []}, {"error_line": "crypto/threads_pthread.c:794:40: error: invalid application of 'sizeof' to incomplete type 'struct stack_traces'", "relevant_repairs": []}, {"error_line": "crypto/threads_pthread.c:797:11: error: invalid use of undefined type 'struct stack_traces'", "relevant_repairs": []}], "repair_source": "dependency", "error_details": ["crypto/threads_pthread.c:747:60: error: 'thread_contention_data' undeclared (first use in this function)\n  747 |     struct stack_traces *traces = CRYPTO_THREAD_get_local(&thread_contention_data);\n      |                                                            ^~~~~~~~~~~~~~~~~~~~~~\ncrypto/threads_pthread.c:747:60: note: each undeclared identifier is reported only once for each function it appears in", "In file included from crypto/threads_pthread.c:34:\ncrypto/threads_pthread.c:749:40: error: invalid application of 'sizeof' to incomplete type 'struct stack_traces'\n  749 |         traces = OPENSSL_zalloc(sizeof(struct stack_traces));\n      |                                        ^~~~~~\ninclude/openssl/crypto.h:104:23: note: in definition of macro 'OPENSSL_zalloc'\n  104 |         CRYPTO_zalloc(num, OPENSSL_FILE, OPENSSL_LINE)\n      |                       ^~~", "crypto/threads_pthread.c:752:11: error: invalid use of undefined type 'struct stack_traces'\n  752 |     traces->lock_depth++;\n      |           ^~\ncrypto/threads_pthread.c: In function 'CRYPTO_THREAD_write_lock':", "crypto/threads_pthread.c:792:60: error: 'thread_contention_data' undeclared (first use in this function)\n  792 |     struct stack_traces *traces = CRYPTO_THREAD_get_local(&thread_contention_data);\n      |                                                            ^~~~~~~~~~~~~~~~~~~~~~", "crypto/threads_pthread.c:794:40: error: invalid application of 'sizeof' to incomplete type 'struct stack_traces'\n  794 |         traces = OPENSSL_zalloc(sizeof(struct stack_traces));\n      |                                        ^~~~~~\ninclude/openssl/crypto.h:104:23: note: in definition of macro 'OPENSSL_zalloc'\n  104 |         CRYPTO_zalloc(num, OPENSSL_FILE, OPENSSL_LINE)\n      |                       ^~~", "crypto/threads_pthread.c:797:11: error: invalid use of undefined type 'struct stack_traces'\n  797 |     traces->lock_depth++;\n      |           ^~\nAt top level:\ncc1: note: unrecognized command-line option '-Wno-unterminated-string-initialization' may have been intended to silence earlier diagnostics"], "compilation_related_paths": {"error_file": ["crypto/threads_pthread.c"]}, "compilation_related_paths_details": [{"error_file": ["crypto/threads_pthread.c"]}, {"error_file": ["crypto/threads_pthread.c"]}, {"error_file": ["crypto/threads_pthread.c"]}, {"error_file": ["crypto/threads_pthread.c"]}, {"error_file": ["crypto/threads_pthread.c"]}, {"error_file": ["crypto/threads_pthread.c"]}]}
{"failure_commit": "d98c44f116490189453ebd91b1a0d0a3f1f21041", "repair_commit": "6c108b9e4590903a94347cb572d24bc8ebe2c753", "error_lines": ["crypto/threads_pthread.c:747:60: error: use of undeclared identifier 'thread_contention_data'", "crypto/threads_pthread.c:749:33: error: invalid application of 'sizeof' to an incomplete type 'struct stack_traces'", "crypto/threads_pthread.c:750:34: error: use of undeclared identifier 'thread_contention_data'", "crypto/threads_pthread.c:752:11: error: incomplete definition of type 'struct stack_traces'", "crypto/threads_pthread.c:792:60: error: use of undeclared identifier 'thread_contention_data'", "crypto/threads_pthread.c:794:33: error: invalid application of 'sizeof' to an incomplete type 'struct stack_traces'", "crypto/threads_pthread.c:795:34: error: use of undeclared identifier 'thread_contention_data'", "crypto/threads_pthread.c:797:11: error: incomplete definition of type 'struct stack_traces'"], "workflow_id": 16050554668, "job_id": [45291945152, 45291945157, 45291945178, 45291945179, 45291945296, 45291945324, 45291945347, 45291945349], "workflow_name": "GitHub CI", "job_name": ["no-shared-macos (macos-14)", "basic_clang", "no-shared-macos (macos-13)", "freebsd-x86_64", "threads_sanitizer", "out-of-readonly-source-and-install-macos (macos-13)", "memory_sanitizer", "out-of-readonly-source-and-install-macos (macos-14)"], "diffs": [{"error_line": "crypto/threads_pthread.c:747:60: error: use of undeclared identifier 'thread_contention_data'", "relevant_repairs": []}, {"error_line": "crypto/threads_pthread.c:749:33: error: invalid application of 'sizeof' to an incomplete type 'struct stack_traces'", "relevant_repairs": []}, {"error_line": "crypto/threads_pthread.c:750:34: error: use of undeclared identifier 'thread_contention_data'", "relevant_repairs": []}, {"error_line": "crypto/threads_pthread.c:752:11: error: incomplete definition of type 'struct stack_traces'", "relevant_repairs": []}, {"error_line": "crypto/threads_pthread.c:792:60: error: use of undeclared identifier 'thread_contention_data'", "relevant_repairs": []}, {"error_line": "crypto/threads_pthread.c:794:33: error: invalid application of 'sizeof' to an incomplete type 'struct stack_traces'", "relevant_repairs": []}, {"error_line": "crypto/threads_pthread.c:795:34: error: use of undeclared identifier 'thread_contention_data'", "relevant_repairs": []}, {"error_line": "crypto/threads_pthread.c:797:11: error: incomplete definition of type 'struct stack_traces'", "relevant_repairs": []}], "repair_source": "dependency", "error_details": ["crypto/threads_pthread.c:747:60: error: use of undeclared identifier 'thread_contention_data'\n    struct stack_traces *traces = CRYPTO_THREAD_get_local(&thread_contention_data);\n                                                           ^", "crypto/threads_pthread.c:749:33: error: invalid application of 'sizeof' to an incomplete type 'struct stack_traces'\n        traces = OPENSSL_zalloc(sizeof(struct stack_traces));\n                                ^     ~~~~~~~~~~~~~~~~~~~~~\ninclude/openssl/crypto.h:104:23: note: expanded from macro 'OPENSSL_zalloc'\n        CRYPTO_zalloc(num, OPENSSL_FILE, OPENSSL_LINE)\n                      ^~~\ncrypto/threads_pthread.c:747:12: note: forward declaration of 'struct stack_traces'\n    struct stack_traces *traces = CRYPTO_THREAD_get_local(&thread_contention_data);\n           ^", "crypto/threads_pthread.c:750:34: error: use of undeclared identifier 'thread_contention_data'\n        CRYPTO_THREAD_set_local(&thread_contention_data, traces);\n                                 ^", "crypto/threads_pthread.c:752:11: error: incomplete definition of type 'struct stack_traces'\n    traces->lock_depth++;\n    ~~~~~~^\ncrypto/threads_pthread.c:747:12: note: forward declaration of 'struct stack_traces'\n    struct stack_traces *traces = CRYPTO_THREAD_get_local(&thread_contention_data);\n           ^", "crypto/threads_pthread.c:792:60: error: use of undeclared identifier 'thread_contention_data'\n    struct stack_traces *traces = CRYPTO_THREAD_get_local(&thread_contention_data);\n                                                           ^", "crypto/threads_pthread.c:794:33: error: invalid application of 'sizeof' to an incomplete type 'struct stack_traces'\n        traces = OPENSSL_zalloc(sizeof(struct stack_traces));\n                                ^     ~~~~~~~~~~~~~~~~~~~~~\ninclude/openssl/crypto.h:104:23: note: expanded from macro 'OPENSSL_zalloc'\n        CRYPTO_zalloc(num, OPENSSL_FILE, OPENSSL_LINE)\n                      ^~~\ncrypto/threads_pthread.c:792:12: note: forward declaration of 'struct stack_traces'\n    struct stack_traces *traces = CRYPTO_THREAD_get_local(&thread_contention_data);\n           ^", "crypto/threads_pthread.c:795:34: error: use of undeclared identifier 'thread_contention_data'\n        CRYPTO_THREAD_set_local(&thread_contention_data, traces);\n                                 ^", "crypto/threads_pthread.c:797:11: error: incomplete definition of type 'struct stack_traces'\n    traces->lock_depth++;\n    ~~~~~~^\ncrypto/threads_pthread.c:792:12: note: forward declaration of 'struct stack_traces'\n    struct stack_traces *traces = CRYPTO_THREAD_get_local(&thread_contention_data);\n           ^"], "compilation_related_paths": {"error_file": ["crypto/threads_pthread.c"]}, "compilation_related_paths_details": [{"error_file": ["crypto/threads_pthread.c"]}, {"error_file": ["crypto/threads_pthread.c"]}, {"error_file": ["crypto/threads_pthread.c"]}, {"error_file": ["crypto/threads_pthread.c"]}, {"error_file": ["crypto/threads_pthread.c"]}, {"error_file": ["crypto/threads_pthread.c"]}, {"error_file": ["crypto/threads_pthread.c"]}, {"error_file": ["crypto/threads_pthread.c"]}]}
{"failure_commit": "a17e2d9c6d2b6741778a80a80a85be84d4b0e9a6", "repair_commit": "de5a619aa015e7c8648e415975e5e2b722b2cbf7", "error_lines": ["apps/s_client.c:945:5: error: unknown type name 'bool'", "apps/s_client.c:945:34: error: 'false' undeclared (first use in this function)", "apps/s_client.c:1553:37: error: 'true' undeclared (first use in this function)"], "workflow_id": 15942361272, "job_id": [44972125549], "workflow_name": "GitHub CI", "job_name": ["minimal"], "diffs": [{"error_line": "apps/s_client.c:945:5: error: unknown type name 'bool'", "relevant_repairs": []}, {"error_line": "apps/s_client.c:945:34: error: 'false' undeclared (first use in this function)", "relevant_repairs": []}, {"error_line": "apps/s_client.c:1553:37: error: 'true' undeclared (first use in this function)", "relevant_repairs": []}], "repair_source": "main_branch", "error_details": ["apps/s_client.c:945:5: error: unknown type name 'bool'\n  945 |     bool record_size_limit_set = false;\n      |     ^~~~\napps/s_client.c:51:1: note: 'bool' is defined in header '<stdbool.h>'; did you forget to '#include <stdbool.h>'?\n   50 | #include \"timeouts.h\"\n  +++ |+#include <stdbool.h>\n   51 | #include \"internal/sockets.h\"", "apps/s_client.c:945:34: error: 'false' undeclared (first use in this function)\n  945 |     bool record_size_limit_set = false;\n      |                                  ^~~~~\napps/s_client.c:945:34: note: 'false' is defined in header '<stdbool.h>'; did you forget to '#include <stdbool.h>'?\napps/s_client.c:945:34: note: each undeclared identifier is reported only once for each function it appears in", "apps/s_client.c:1553:37: error: 'true' undeclared (first use in this function)\n 1553 |             record_size_limit_set = true;\n      |                                     ^~~~\napps/s_client.c:1553:37: note: 'true' is defined in header '<stdbool.h>'; did you forget to '#include <stdbool.h>'?\nAt top level:\ncc1: note: unrecognized command-line option '-Wno-unterminated-string-initialization' may have been intended to silence earlier diagnostics"], "compilation_related_paths": {"error_file": ["apps/s_client.c"]}, "compilation_related_paths_details": [{"error_file": ["apps/s_client.c"]}, {"error_file": ["apps/s_client.c"]}, {"error_file": ["apps/s_client.c"]}]}
{"failure_commit": "4b6f6604d8ad09c1fc0dcd3f122fd6dfa4c02ace", "repair_commit": "e1bccde1a76cb3c2b9204093083b8704b81ac37f", "error_lines": ["ssl/record/rec_layer_d1.c:618:26: error: use of undeclared identifier 'sc'"], "workflow_id": 15446358946, "job_id": [43477122273, 43477181806, 43477249381, 43477261249, 43477265157, 43477284634, 43477293929, 43477296405], "workflow_name": "GitHub CI", "job_name": ["basic_clang", "freebsd-x86_64", "memory_sanitizer", "no-shared-macos (macos-13)", "no-shared-macos (macos-14)", "threads_sanitizer", "out-of-readonly-source-and-install-macos (macos-13)", "out-of-readonly-source-and-install-macos (macos-14)"], "diffs": [{"error_line": "ssl/record/rec_layer_d1.c:618:26: error: use of undeclared identifier 'sc'", "relevant_repairs": []}], "repair_source": "same_branch", "error_details": ["ssl/record/rec_layer_d1.c:618:26: error: use of undeclared identifier 'sc'\n  618 |                 SSLfatal(sc, SSL_AD_HANDSHAKE_FAILURE, SSL_R_NO_RENEGOTIATION);\n      |                          ^"], "compilation_related_paths": {"error_file": ["ssl/record/rec_layer_d1.c"]}, "compilation_related_paths_details": [{"error_file": ["ssl/record/rec_layer_d1.c"]}]}
{"failure_commit": "4b6f6604d8ad09c1fc0dcd3f122fd6dfa4c02ace", "repair_commit": "e1bccde1a76cb3c2b9204093083b8704b81ac37f", "error_lines": ["ssl/record/rec_layer_d1.c:618:26: error: 'sc' undeclared (first use in this function); did you mean 's'?"], "workflow_id": 15446358946, "job_id": [43477134293, 43477145239, 43477207177, 43477214724, 43477221284, 43477227139, 43477233376, 43477237873, 43477242273, 43477246157, 43477253150, 43477257234, 43477271960, 43477278151, 43477287514, 43477290399], "workflow_name": "GitHub CI", "job_name": ["basic_gcc", "check-ansi", "linux-arm64", "minimal", "no-deprecated", "address_ub_sanitizer", "buildtest", "enable_non-default_options", "fips_and_ktls", "legacy", "no-legacy", "non-caching", "no-shared-ubuntu", "out-of-readonly-source-and-install-ubuntu", "external-test-pyca (1.51.0, 3.9)", "external-tests"], "diffs": [{"error_line": "ssl/record/rec_layer_d1.c:618:26: error: 'sc' undeclared (first use in this function); did you mean 's'?", "relevant_repairs": []}], "repair_source": "same_branch", "error_details": ["In file included from ssl/record/../ssl_local.h:31,\nssl/record/rec_layer_d1.c:618:26: error: 'sc' undeclared (first use in this function); did you mean 's'?\n  618 |                 SSLfatal(sc, SSL_AD_HANDSHAKE_FAILURE, SSL_R_NO_RENEGOTIATION);\n      |                          ^~\nssl/record/../statem/statem.h:139:44: note: in definition of macro 'SSLfatal'\n  139 | # define SSLfatal(s, al, r) SSLfatal_data((s), (al), (r), NULL)\n      |                                            ^\nssl/record/rec_layer_d1.c:618:26: note: each undeclared identifier is reported only once for each function it appears in\n  618 |                 SSLfatal(sc, SSL_AD_HANDSHAKE_FAILURE, SSL_R_NO_RENEGOTIATION);\n      |                          ^~\nssl/record/../statem/statem.h:139:44: note: in definition of macro 'SSLfatal'\n  139 | # define SSLfatal(s, al, r) SSLfatal_data((s), (al), (r), NULL)\n      |                                            ^\nAt top level:\ncc1: note: unrecognized command-line option '-Wno-unterminated-string-initialization' may have been intended to silence earlier diagnostics"], "compilation_related_paths": {"error_file": ["ssl/record/rec_layer_d1.c"]}, "compilation_related_paths_details": [{"error_file": ["ssl/record/rec_layer_d1.c"]}]}
{"failure_commit": "c489a6205ec25528d4ab692694571af75b35879e", "repair_commit": "f32e8a48d20c448a6706e63ce920afbc4b40b203", "error_lines": ["include/internal/numbers.h:109:1: error: C++ style comments are not allowed in ISO C90"], "workflow_id": 15324203276, "job_id": [43118303014], "workflow_name": "GitHub CI", "job_name": ["check-ansi"], "diffs": [{"error_line": "include/internal/numbers.h:109:1: error: C++ style comments are not allowed in ISO C90", "relevant_repairs": []}], "repair_source": "dependency", "error_details": ["In file included from include/internal/e_os.h:18,\ninclude/internal/numbers.h:109:1: error: C++ style comments are not allowed in ISO C90\n  109 | // Fix for cross compiling 64-bit PowerPC on OS X 10.4\n      | ^"], "compilation_related_paths": {"error_file": ["include/internal/numbers.h"]}, "compilation_related_paths_details": [{"error_file": ["include/internal/numbers.h"]}]}
{"failure_commit": "322872068199942a581d26ed72e6dde6ac2d75f2", "repair_commit": "7cc69ec4cf575232040ac776f405fb31f030e59a", "error_lines": ["ssl/s3_lib.c:5044:22: error: 'PROV_R_FAILED_DURING_DERIVATION' undeclared (first use in this function)"], "workflow_id": 14970760878, "job_id": [42050978585, 42050980513, 42050985413, 42050989062, 42050990754, 42050991545, 42050992358, 42050992786, 42050993187, 42050993481, 42050993919, 42050994738, 42050995113, 42050995817, 42050996878, 42050997327, 42050997797], "workflow_name": "GitHub CI", "job_name": ["check-ansi", "basic_gcc", "linux-arm64", "minimal", "no-deprecated", "no-shared-ubuntu", "address_ub_sanitizer", "enable_non-default_options", "full_featured", "fuzz_tests", "legacy", "no-legacy", "non-caching", "out-of-readonly-source-and-install-ubuntu", "external-tests-misc", "external-tests-providers", "external-tests-pyca (3.9)"], "diffs": [{"error_line": "ssl/s3_lib.c:5044:22: error: 'PROV_R_FAILED_DURING_DERIVATION' undeclared (first use in this function)", "relevant_repairs": []}], "repair_source": "main_branch", "error_details": ["In file included from ssl/ssl_local.h:30,\nssl/s3_lib.c:5044:22: error: 'PROV_R_FAILED_DURING_DERIVATION' undeclared (first use in this function)\n 5044 |                      PROV_R_FAILED_DURING_DERIVATION);\n      |                      ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~\ninclude/internal/statem.h:148:55: note: in definition of macro 'SSLfatal'\n  148 | # define SSLfatal(s, al, r) SSLfatal_data((s), (al), (r), NULL)\n      |                                                       ^\nssl/s3_lib.c:5044:22: note: each undeclared identifier is reported only once for each function it appears in\n 5044 |                      PROV_R_FAILED_DURING_DERIVATION);\n      |                      ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~\ninclude/internal/statem.h:148:55: note: in definition of macro 'SSLfatal'\n  148 | # define SSLfatal(s, al, r) SSLfatal_data((s), (al), (r), NULL)\n      |                                                       ^\nAt top level:\ncc1: note: unrecognized command-line option '-Wno-unterminated-string-initialization' may have been intended to silence earlier diagnostics"], "compilation_related_paths": {"error_file": ["ssl/s3_lib.c"]}, "compilation_related_paths_details": [{"error_file": ["ssl/s3_lib.c"]}]}
{"failure_commit": "322872068199942a581d26ed72e6dde6ac2d75f2", "repair_commit": "7cc69ec4cf575232040ac776f405fb31f030e59a", "error_lines": ["ssl/s3_lib.c:5044:22: error: use of undeclared identifier 'PROV_R_FAILED_DURING_DERIVATION'"], "workflow_id": 14970760878, "job_id": [42050979620, 42050982333, 42050991188, 42050991904, 42050994361, 42050995437, 42050996159, 42050996524], "workflow_name": "GitHub CI", "job_name": ["basic_clang", "freebsd-x86_64", "no-shared-macos (macos-13)", "no-shared-macos (macos-14)", "memory_sanitizer", "out-of-readonly-source-and-install-macos (macos-13)", "out-of-readonly-source-and-install-macos (macos-14)", "threads_sanitizer"], "diffs": [{"error_line": "ssl/s3_lib.c:5044:22: error: use of undeclared identifier 'PROV_R_FAILED_DURING_DERIVATION'", "relevant_repairs": []}], "repair_source": "main_branch", "error_details": ["ssl/s3_lib.c:5044:22: error: use of undeclared identifier 'PROV_R_FAILED_DURING_DERIVATION'\n 5044 |                      PROV_R_FAILED_DURING_DERIVATION);\n      |                      ^"], "compilation_related_paths": {"error_file": ["ssl/s3_lib.c"]}, "compilation_related_paths_details": [{"error_file": ["ssl/s3_lib.c"]}]}
